doozerlib/cli/__main__.py

```diff
@@ -1518,6 +1522,132 @@ def config_read_group(runtime, key, as_len, as_yaml, permit_missing_group, defau
     print(str(value))
 
 
+def get_releases(group) -> dict:
+    """
+    Uses GitHub API to fecth releases.yaml from openshift-eng/ocp-build-data for a given group
+
+    Parses the file and returns it as a dictionary
+    """
+
+    if not os.environ.get('GITHUB_TOKEN', None):
+        raise DoozerFatalError('A GITHUB_TOKEN environment variable must be defined!')
+    github_token = os.environ['GITHUB_TOKEN']
+
+    api = GhApi(owner='openshift-eng', repo='ocp-build-data', token=github_token)
+    files = api.list_files(branch=group)
+    releases_yaml = files.get('releases.yml')
+    blob = api.git.get_blob(file_sha=releases_yaml['sha'])
+    return yaml.safe_load(base64.b64decode(blob['content']))
+
+
+@cli.command("config:read-releases", short_help="Output aspects of releases.yml")
+@click.option("--length", "as_len", default=False, is_flag=True, help='Print number of assemblies defined for group')
+@click.option("--yaml", "as_yaml", default=False, is_flag=True, help='Print results in a yaml block')
+@click.option("--out-file", help="Output contents to a file instead of stdout", default=None)
+@pass_runtime
+def config_read_releases(runtime, as_len, as_yaml, out_file):
+    """
+    Read data from releases.yaml for given group and key.
+    If key is not specified, the entire release data structure will be output.
+
+    Usage:
+
+    $ doozer --group=openshift-4.14 config:read-releases
+
+    $ doozer --group=openshift-4.14 config:read-releases --yaml
+
+    $ doozer --group=openshift-4.13 config:read-releases --length
+
+    $ doozer --group=openshift-4.13 config:read-releases --yaml --out-file /tmp/out.yaml
+    """
+
+    content = get_releases(runtime.group)
+
+    if as_len:
+        output = len(content['releases'])
+    elif as_yaml:
+        output = yaml.safe_dump(content)
+    else:
+        output = content
+
+    if out_file:
+        # just in case
+        out_file = os.path.expanduser(out_file)
+        try:
+            with io.open(out_file, 'w', encoding="utf-8") as f:
+                f.write(str(output))
+        except PermissionError:
+            click.echo(f'Permission denied: could not write to {out_file}')
+            sys.exit(1)
+
+    else:
+        click.echo(output)
+
+
+@cli.command("config:read-assemblies", short_help="Output aspects of assemblies defined in releases.yml")
```

Comment:
Let's call it `config:read-assembly` since it only returns the data for a single assembly.

Reply:
Done in 0fdf7acb35c3b590f0cdf033484ee9495a2e9025